### PR TITLE
IGNITE-21974: Extend test coverage for SQL F221(Explicit defaults)

### PR DESCRIPTION
**Pull request by lowka**
Adds java tests for DROP default.

https://issues.apache.org/jira/browse/IGNITE-21974

--- 
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java [hunk 1]
```diff
@@ -596,12 +598,55 @@ private void checkDefaultValue(List<DefaultValueArg> args) {
 
     @Test
     public void testCheckNullValueErrorMessageForColumnWithDefaultValue() {
+        // F221: Explicit defaults
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java:601]: Let's say more explicitly, like a SQL Standard 2016 feature F221

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java [hunk 2]
```diff
@@ -596,12 +598,55 @@ private void checkDefaultValue(List<DefaultValueArg> args) {
...
 
         assertThrowsSqlException(Sql.CONSTRAINT_VIOLATION_ERR, expectedMessage, () -> sql("INSERT INTO tbl (key, val) VALUES (NULL,'AA')"));
     }
+    
+    // UPDATE set x = DEFAULT is not supported in parser
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-21462")
+    @Test
+    public void testUpdateAllowsDefault() {
+        // F221: Explicit defaults
+        for (var arg : defaultValueArgs().collect(Collectors.toList())) {
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java:614]: var is forbidden in such context

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java [hunk 3]
```diff
@@ -596,12 +598,55 @@ private void checkDefaultValue(List<DefaultValueArg> args) {
...
+            } finally {
+                sql("DROP TABLE IF EXISTS test");
+            }
+        }
+    }
+
+    @Test
+    public void testDropDefault() {
+        // F221: Explicit defaults
+        for (var arg : defaultValueArgs().collect(Collectors.toList())) {
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java:630]: Seems need to extend type coverage

Comment (lowka) [reply modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java:629]: All types have a NOT NULL variant.

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java (hunk 2 above)
Comment (lowka) [reply modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java]: Fixed.

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java (hunk 1 above)
Comment (lowka) [reply modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java]: Done.

**Review by ygerzhedovich — APPROVED** [review]

**Pull request merged**